frontend/test/pytest/test_jax_dynamic_api.py

```diff
@@ -1139,5 +1139,48 @@ def f(i, _):
     assert r == 3
 
 
+def test_abstracted_axis_no_recompilation():
+    """Test that a function that does not need recompilation can be executed a second time"""
+
+    @qml.qjit(abstracted_axes=(("n",), ()))
+    @qml.qnode(qml.device("lightning.qubit", wires=2))
+    def circuit(x1, x2):
+
+        @qml.for_loop(0, jnp.shape(x1)[0], 1)
+        def loop_block(i):
+            qml.Hadamard(0)
+            qml.RX(x1[i], 0)
+            qml.CNOT(wires=[0, 1])
+            qml.Hadamard(1)
```

Comment:
To save testing time in CI, why don't we remove the static gates that don't depend on the arguments?

Reply:
Done :+1: